### DISCUSSION TITLE
Default Language Selection in Translations

### DISCUSSION THREAD

**Discussion by luismts** [directus/directus#21052] [Feature Requests]
### Summary

This feature proposes enhancing the translations functionality in Directus by automatically selecting the default language when creating a new translation for a field name. This aims to streamline the translation process and reduce user effort.

### Basic Example

When creating a translation for a field name, the default language (e.g., Spanish) should be pre-selected in the language dropdown by default. 

![image](https://github.com/directus/directus/assets/20643876/5f741838-d266-4f78-a772-723bcc103558)

The current behavior is the English is selected by default.
![image](https://github.com/directus/directus/assets/20643876/56eb99ae-fe89-4f59-bf8c-fc7dea2e4c0c)

### Motivation

The motivation behind this feature is to improve the user experience when managing translations. Currently, when creating a new translation, users need to manually select the default language each time. This enhancement aims to save time, reduce the likelihood of errors, and create a more intuitive workflow for users dealing with multilingual content.

### Detailed Design

- When a user clicks on "Create Translation" for a field name, the language dropdown should default to the language of the field being translated.
- If the default language is changed for the field, the language dropdown should dynamically update accordingly.
- The behavior should be consistent across various translation contexts, ensuring a seamless experience for users.

### Must Have:
- Automatic pre-selection of the default language in the translations dropdown.
- Dynamic updates to the language dropdown based on changes to the default language.

### Should Have:
- Clear visual indicators to highlight the default language in the translations interface.

### Could Have:
- Configuration options to customize the default language selection behavior.

### Won't Have:
- Integration with every conceivable language-related customization.

### Drawbacks

- Increased complexity in managing dynamic language dropdown behavior.
- Potential for users to overlook or misunderstand the default language selection.

### Alternatives

- Users can continue with the current manual selection process.
- Development of third-party plugins or extensions for customization.

### Adoption Strategy

- Provide clear documentation and tutorials for users to understand and leverage the new default language selection.
- Implement backward compatibility where possible to minimize breaking changes.
- Develop a migration path for existing Directus applications to accommodate the new behavior.

### Unresolved Questions

- Detailed design considerations for maintaining consistency in the translation interface.
- Exploration of additional customization options based on user feedback.
- Impact on performance with the dynamic update of the language dropdown.

**Comment by luismts** [comment]
### Enhancements and Desired Features for Directus
I have created a thread with a list of features and enhancements that could further drive the utility and versatility of Directus.

You can see it [here](https://gist.github.com/luismts/ed40fb3ce57d0697f604a8ef59a2c875).

**Comment by paescuj** [comment]
This has been implemented in #21065 and will be available in the next release 👍
